cli/build.gradle.kts

```diff
@@ -44,6 +44,11 @@ tasks {
     }
 
     "test" {
+        dependsOn("shadowJar")
         dependsOn(":analyzer:assemble")
     }
+
+    "publishShadowPublicationToMavenLocal" {
+        dependsOn("jar")
+    }
```

Comment:
Can you clarify these please?

Reply:
Granted, for this task, it only happens when you use `publishToMavenLocal`. But, in general:

There's a new warning output when built with Gradle 7+. It is explained in the following links
<https://docs.gradle.org/7.0/release-notes.html#build-reliability-improvements> and <https://docs.gradle.org/7.0/userguide/validation_problems.html#implicit_dependency>.

It is of the form:

Execution optimizations have been disabled for task ':task' to ensure correctness due to the following reasons:
  - Gradle detected a problem with the following location: 'location'. Reason: Task ':task' uses this output of task ':otherTask' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.0/userguide/validation_problems.html#implicit_dependency for more details about this problem.

Reply:
Ok that makes sense. Can we add a sentence or so as comments describing why this extra code was needed please